qiskit/providers/basic_provider/basic_provider_tools.py

```diff
@@ -40,42 +63,93 @@ def single_gate_matrix(gate: str, params: list[float] | None = None) -> np.ndarr
     """
     if params is None:
         params = []
-
-    if gate == "U":
+    if gate in ("U", "u"):
         gc = gates.UGate
+    elif gate == "u1":
+        gc = gates.U1Gate
+    elif gate == "u2":
+        gc = gates.U2Gate
     elif gate == "u3":
         gc = gates.U3Gate
     elif gate == "h":
         gc = gates.HGate
-    elif gate == "u":
-        gc = gates.UGate
     elif gate == "p":
         gc = gates.PhaseGate
```

Comment:
If we're going to add this many new gates, please can we switch to using a dict lookup to retrieve the gate classes?

Reply:
done in 5cddee7572e4a369a0f26c5d0aaa4fe0c6dd5d36